items/metadata/metadata.js

```diff
@@ -74,9 +74,45 @@ const removeMetadata = function (itemOrName, namespace) {
   };
 };
 
+// TODO: Move implementation to openHAB Core
+/**
+ * Get all metadata from a given Item.
+ *
+ * @example
+ * var meta = items.metadata.getItemMetadata('Kitchen_Light');
+ * var namespaces = Array.from(meta.keys());
+ * var stateDescription = meta.get('stateDescription');
+ *
+ * @memberof items.metadata
+ * @param {string} itemName the name of the Item
+ * @returns {Map<string, {configuration: object, value: string}>} Map of metadata
+ */
+const getItemMetadata = function (itemName) {
```

Comment:
So we have `item.getMetaData(namespace)`, `metadata.getMetadata(itemName, nameSpace)`, and `metadata. getItemMetadata(itemName)`.  I'm not entirely clear on the the purpose of the two latter functions?  They sound very similar.

Reply:
`Item.getMetadate(namespace)` and `items.metadata.getMetadata(itemOrName, namespace)` are the same functions. The prior is just a "shortcut" on the Item object so you don't need to call the latter method with the Item or Item name.

`getItemMetadata(itemOrName)` however gets ALL metadata an Item has, not only one metadata namespace like the two functions above.

Reply:
Right, so i'm wondering if we should try and combine `items.metadata.getMetadata` and `metadata. getItemMetadata` into single function, maybe make the namespace optional?  Does that complicate the return structure too much ?

Reply:
Sounds like a great idea, we could switch to return an object in case we want to get all metadata. 
I’m just not sure yet how to document the return, but I guess this should be possible using by using some JSDoc „magic“. But type definitions/auto-completion will get less exact I guess.

Reply:
I was able to combine the get and remove metdata methods each into a single one that accepts the namespace as an optional parameter. Return type documentation seems to be no real problem, I have checked the generated type defs, they look good as well.

Reply:
Thanks.  I see now, so it returns
```js
{
 ns: {
 value: v,
 configuration: c
},
...more ns entries ...
}
```
or
```js
{
 value: v,
 configuration: c
}
```

Depending on parameter.  I can see the need to for two methods ( i was not sure of the return structure before).   I wonder then if we should keep two methods and call them  `getAllMetadata()` and `getMetadata(namespace)` (or something similar, i'm not great at naming things).  This is where function overloading comes in handy in Java, but i guess in JS dynamic return types are probably equivalent ?    I'm really good either way, wdyt?

Reply:
I put some work in the documentation of the return type in JSDoc and the type defs and I think keeping these "unified" methods therefore is okay :+1: 

How is your review going?

Reply:
Good, on a plane with spotty internet :-)

Reply:
Ooh. You don't need to be on a plane for spotty internet ;-) 
At my home it is often too much for the internet when it's evening and everyone wants to stream something ...